### PR TITLE
osd/OSDMap: Uncomment code to enable private default constructors

### PR DESCRIPTION
**Pull request by badone**
C++11 is here so let's enable this code.

Signed-off-by: Brad Hubbard <bhubbard@redhat.com>

### CODE REVIEW

**Comment by badone** [comment]
I swear I built and tested this locally with no problems. Looking into this.

**Comment by badone** [comment]
@liewegas and @athanatos how does this look?

File: src/osd/OSDMap.h [hunk 1]
```diff
@@ -275,12 +275,10 @@ class OSDMap {
   }
 
   // no copying
-  /* oh, how i long for c++11...
 private:
   OSDMap(const OSDMap& other) = default;
```
Comment (tchaikov) [inline src/osd/OSDMap.h]: if we want to disable the default copy ctor and assign operator, we might want to put 
```c++
OSDMap(const OSDMap& other) = delete;
OSDMap& operator=(const OSDMap& other) = delete;
```
instead, and probably remove the `private:` before them also.

Comment (badone) [inline src/osd/OSDMap.h]: Yes Kefu, I believe the intent was that they would only be usable within the OSDMap class and unusable from outside of that class but I'll let @liewegas and/or @athanatos comment on that. Happy to make the change if required of course and thanks for taking a look.

Comment (liewegas) [inline src/osd/OSDMap.h]: if everything builds with = delete, let's do that.  I forget if there is a private user.

File: src/test/encoding/types.h [hunk 2]
```diff
@@ -41,7 +41,7 @@ TYPE_FEATUREFUL(entity_inst_t)
 #include "osd/OSDMap.h"
 TYPE(osd_info_t)
 TYPE(osd_xinfo_t)
-TYPE_FEATUREFUL_STRAYDATA(OSDMap)
+TYPE_FEATUREFUL_NOCOPY(OSDMap)
```
Comment (liewegas) [inline src/test/encoding/types.h]: make sure test/encoding/readable.sh works with this change...

Comment (badone) [inline src/test/encoding/types.h]: Sure, that was my original suggestion as I thought that was your intent originally but Sam talked me out of it :) Another change coming up...

Comment (badone) [inline src/test/encoding/types.h]: "= delete" I mean

Comment (badone) [inline src/test/encoding/types.h]: Alas, it doesn't build since we use assignment in deepish_copy_from(). Let me know which way you want to jump.

Comment (badone) [inline src/test/encoding/types.h]: test/encoding/readable.sh works with the current changes BTW.

**Comment by liewegas** [comment]
private is fine then!

**Comment by badone** [comment]
@tchaikov happy with that?

**Comment by badone** [comment]
Sorry, I merged this before it passed testing.

**Comment by liewegas** [comment]
No worries, it passed my testing run.

**Pull request merged**